app/src/main/java/ru/prometeydev/movie/ui/base/Event.kt

```diff
@@ -0,0 +1,29 @@
+package ru.prometeydev.movie.ui.base
+
+data class Event<out T>(
```

Comment:
Check the sealed classes.

sealed class Result<T> {
  class Error(val errorMessage: String?) : Result<Nothing>
 ....
}

Reply:
Then you don't need Status enum at all